src/Slider.test.tsx

```diff
@@ -127,24 +128,32 @@ describe('UpsellSlider', () => {
             const clickSpy = jest.fn();
 
             render(<Slider>
-                <span data-testid="1" onClick={clickSpy}/>
+                <span key={ 1 }  data-testid="1" onClick={clickSpy}/>
             </Slider>);
 
             const scrollElement = screen.getByRole('list');
 
-            // eslint-disable-next-line testing-library/prefer-user-event
-            fireEvent.mouseDown(scrollElement);
-            // eslint-disable-next-line testing-library/prefer-user-event
-            fireEvent.mouseMove(scrollElement, { clientX: 100, clientY: 0 });
-            // eslint-disable-next-line testing-library/prefer-user-event
-            fireEvent.mouseUp(scrollElement);
+            act(() => {
+                // eslint-disable-next-line testing-library/prefer-user-event
+                fireEvent.mouseDown(scrollElement);
+                // eslint-disable-next-line testing-library/prefer-user-event
+                fireEvent.mouseMove(scrollElement, { clientX: 100, clientY: 0 });
+                // eslint-disable-next-line testing-library/prefer-user-event
+                fireEvent.mouseUp(scrollElement);
+            });
 
             // This click is normally triggered when releasing the mouse after scrolling
-            await userEvent.click(scrollElement);
+            act(() => {
+                userEvent.click(scrollElement);
+            });
 
-            await userEvent.click(screen.getByTestId('1'));
+            act(() => {
+                userEvent.click(screen.getByTestId('1'));
+            });
 
-            expect(clickSpy).toHaveBeenCalled();
+            await waitFor(() => {
```

Comment:
I take it the test fails if we don't use the waitFor?

Reply:
Yes, due the added act for handling the forwardRef there is a little timing added.